src/vsr/replica.zig

```diff
@@ -3177,6 +3158,11 @@
                 .checkpoint_client_replies => {
                     self.client_replies.checkpoint(commit_op_checkpoint_client_replies_callback);
                 },
+                .checkpoint_client_sessions => {
+                    self.client_sessions_checkpoint.size =
+                        self.client_sessions.encode(self.client_sessions_checkpoint.buffer);
```

Comment:
I am wondering if perhaps there's some way to assert that _both_ size and buffer are correctly set
at the same time? But I don't see any.

Reply:
Yeah, the API is a little awkward. :thinking: